#VRML_SIM R2023b utf8

EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackground.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/backgrounds/protos/TexturedBackgroundLight.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/floors/protos/RectangleArena.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/objects/factory/containers/protos/CardboardBox.proto"
EXTERNPROTO "https://raw.githubusercontent.com/cyberbotics/webots/R2023b/projects/robots/gctronic/e-puck/protos/E-puck.proto"

WorldInfo {
}
Viewpoint {
  orientation -0.5773502691896258 0.5773502691896258 0.5773502691896258 2.0944
  position 0.08216424119573437 -0.3760647542859747 1.3634962616931656
  follow "e-puck"
}
TexturedBackground {
  texture "music_hall"
}
TexturedBackgroundLight {
  texture "music_hall"
}
RectangleArena {
  floorSize 6 2
}
CardboardBox {
  translation -0.26 0.04 0.0496067
  rotation -0.5294041580744199 -0.42352332645953095 -0.7350913068171655 8.05293337715217e-17
  size 0.1 0.2 0.1
  mass 1
}
E-puck {
  translation -0.08491266509076324 -0.3600000696927938 -5.594327555989362e-05
  rotation -2.497781821249441e-06 -0.999999565074598 -0.000932654478205783 0.0012312470900508745
  name "Objek tak dikenal"
  controller "objekgerak"
  version "2"
}
E-puck {
  translation -0.349387 -0.36 -0.000118532
  rotation 2.4022693302037264e-06 0.9999999999903301 3.6836618062179104e-06 0.0009096376307042178
  controller "FIXFuzzy2"
  groundSensorsSlot [
    DistanceSensor {
      translation 0 0 0.03
      children [
        DistanceSensor {
          translation 0 0.02 0
          rotation 0 0 1 0.261799
          children [
            DistanceSensor {
              translation -0.0103526 -0.0386371 0
              rotation 0 0 1 -0.523595307179586
              name "ps11"
              type "laser"
            }
          ]
          name "ps9"
          type "laser"
        }
      ]
      name "ps10"
      type "sonar"
    }
  ]
}
